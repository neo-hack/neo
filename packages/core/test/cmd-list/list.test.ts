import {
  execNeo,
  mockLockFile,
  storeDir,
} from '../helpers'

beforeAll(async () => {
  await mockLockFile()
})

describe('command list', () => {
  it('should contain all templates in store', async () => {
    const { stdout } = await execNeo(['list', '--store-dir', storeDir])
    expect(stdout).toMatchSnapshot()
  })
  it('should show template list filter by preset', async () => {
    const { stdout } = await execNeo(['list', '--store-dir', storeDir, '--preset', 'neo'])
    expect(stdout).toMatchSnapshot()
  })
  it('should show empty template list with wrong preset', async () => {
    const { stdout } = await execNeo(['list', '--store-dir', storeDir, '--preset', 'aiou'])
    expect(stdout).toMatchSnapshot()
  })
})
