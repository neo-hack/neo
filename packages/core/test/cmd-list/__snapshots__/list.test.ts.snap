// Vitest Snapshot v1

exports[`command list > should contain all templates in store 1`] = `
"NEO v3.5.3 (​https://neo-docs.netlify.app​)

[log] Note: cached, uncached

[log] Found 6 templates:

[log] neo:
[log] @aiou/webext-template
bin-template#bundle
bin-template#unbundle
ts-lib-template (https://github.com/egoist/ts-lib-starter)
[log]
[log] demo:
[log] ts-lib-template (@aiou/ts-lib-template)
[log]
[log] Not in preset:
[log] react-template"
`;

exports[`command list > should show empty template list with wrong preset 1`] = `
"NEO v3.5.3 (​https://neo-docs.netlify.app​)

[log] No templates..."
`;

exports[`command list > should show template list filter by preset 1`] = `
"NEO v3.5.3 (​https://neo-docs.netlify.app​)

[log] Note: cached, uncached

[log] Found 4 templates:

[log] @aiou/webext-template
bin-template#bundle
bin-template#unbundle
ts-lib-template"
`;
